src/jit/codegenxarch.cpp

```diff
@@ -4247,241 +4247,133 @@ instruction CodeGen::genGetInsForOper(genTreeOps oper, var_types type)
     return ins;
 }
 
-/** Generates the code sequence for a GenTree node that
- * represents a bit shift or rotate operation (<<, >>, >>>, rol, ror).
- *
- * Arguments: operand:  the value to be shifted or rotated by shiftBy bits.
- *            shiftBy:  the number of bits to shift or rotate the operand.
- *            parent:   the actual bitshift node (that specifies the
- *                      type of bitshift to perform.
- *
- * Preconditions:    a) All GenTrees are register allocated.
- *                   b) Either shiftBy is a contained constant or
- *                      it's an expression sitting in RCX.
- */
-void CodeGen::genCodeForShift(GenTreePtr operand, GenTreePtr shiftBy,
+
+//------------------------------------------------------------------------
+// genCodeForShift: Generates the code sequence for a GenTree node that
+// represents a bit shift or rotate operation (<<, >>, >>>, rol, ror).
+//
+// Arguments:
+//    operand:  the value to be shifted or rotated by shiftBy bits.
+//    shiftBy:  the number of bits to shift or rotate the operand.
+//    parent:   the actual bitshift node (that specifies the
+//              type of bitshift to perform).
+//
+// Assumptions:
+//    a) All GenTrees are register allocated.
+//    b) Either shiftBy is a contained constant or it's an expression sitting in RCX.
+//
+void CodeGen::genCodeForShift(GenTreePtr operand,
+                              GenTreePtr shiftBy,
                               GenTreePtr parent)
```

Comment:
Does this routine need 3 params or just 'parent' param?
if parent is shift oper node. Its op1 is operand and op2 is shiftBy.

Since you are editing this, please change this  as well.